daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs

```diff
@@ -635,10 +636,10 @@ sanitizeToken tok
   | otherwise = "Bearer " <> tok
 
 -- | Report on Ledger Use.
-runLedgerMeteringReport :: LedgerFlags -> IsoTime -> Maybe IsoTime -> Maybe ApplicationId -> Bool -> IO ()
+runLedgerMeteringReport :: LedgerFlags -> Day -> Maybe Day -> Maybe ApplicationId -> Bool -> IO ()
 runLedgerMeteringReport flags fromIso toIso application compactOutput = do
     args <- getDefaultArgs flags
-    report <- meteringReport args (L.isoTimeToTimestamp fromIso) (fmap L.isoTimeToTimestamp toIso) application
+    report <- meteringReport args (L.utcDayToTimestamp fromIso) (fmap L.utcDayToTimestamp toIso) application
```

Comment:
So this looks like the gRPC service still accepts time stamps? Does it error ouf if the time is not 00:00:00?

Reply:
This seems weird. Why would we cripple daml-helper but not the grpc service? Either we support this in which case we should expose it in daml-helper or we don’t support it in which case it should be rejected by the grpc service.

Reply:
The primary use case for the helper is to support billing which will be UTC day based and supporting anything other than this may lead to confusion is the thinking.

Reply:
I’m not keen on this. We generally try to map daml ledger commands pretty much one to one to ledger API services. I think the restriction is fine but then this should be a ledger API restrictions not a daml helper restriction.

Reply:
Alternatively maybe add a special mode for calculating billing that enforces this and keep the main metering mode as a 1:1 mapping.

Reply:
as discussed, let’s go with this for now and make it more flexible later.

Reply:
As discussed https://digitalasset.atlassian.net/browse/DPP-936 raised to track this.